docs/01-infrastructure-os.md

```diff
@@ -0,0 +1,195 @@
+# Cloud Infrastructure Provisioning - OpenStack
+
+This lab will walk you through provisioning the compute instances required for running a H/A Kubernetes cluster. A total of 6 virtual machines will be created.
+
+After completing this guide you should have the following compute instances:
+
+```
+openstack server list
+```
+
+````
++--------------------------------------+-----------------+-----------+----------------------------------------+---------------------+
+| ID                                   | Name            | Status    | Networks                               | Image Name          |
++--------------------------------------+-----------------+-----------+----------------------------------------+---------------------+
+| 17da9ba7-a0c3-415a-9fe2-b2729d4ba3da | worker2         | ACTIVE    | kubernetes=10.240.0.22                 | ubuntu-16.04        |
+| d52281ba-0a76-4abf-addb-cd56c79d3f1d | worker1         | ACTIVE    | kubernetes=10.240.0.21                 | ubuntu-16.04        |
+| f44c0c77-9810-4cf4-977e-45dafbe87074 | worker0         | ACTIVE    | kubernetes=10.240.0.20                 | ubuntu-16.04        |
+| 96e690b4-e8cb-4733-aa1d-5262106181a2 | controller2     | ACTIVE    | kubernetes=10.240.0.12                 | ubuntu-16.04        |
+| d69f09c1-00e5-465a-831c-446206461d28 | controller1     | ACTIVE    | kubernetes=10.240.0.11                 | ubuntu-16.04        |
+| 80fc744c-d20e-4f24-9b10-c8a26ffbade3 | controller0     | ACTIVE    | kubernetes=10.240.0.10, 169.45.x.x     | ubuntu-16.04        |
++--------------------------------------+-----------------+-----------+----------------------------------------+---------------------+
+````
+
+> All machines will be provisioned with fixed private IP addresses to simplify the bootstrap process.
+
+To make our Kubernetes control plane remotely accessible, a Floating IP address will be  assigned to one of the Kubernetes controllers. You can also assign floating IPs to all the nodes. In this example we're going to assign an IP to controller0 and use that to access the remailing nodes.
+
+## Networking
+
+
+Create a Kubernetes network:
+
+```
```

Comment:
This won't work on all clouds, which may not have provide access to creating networks. For instance on OVH, these commands led to permissions denied.